README
NAME
    DBIx::Class::InflateColumn::Serializer::JSYNC - Basic JSON Object
    Serialization Support for DBIx::Class.

VERSION
    version 0.002000

DESCRIPTION
    This is basically the only serialization backend I could find that
    wasn't "Dumper()", and actually seemed to work with arbitrary "bless()"

        package Foo::Result::Thing;
        __PACKAGE__->load_components('InflateColumn::Serializer', 'Core');
        __PACKAGE__->table('thing');

        ....

        __PACKAGE__->add_column(
            data => {
                data_type => 'text',
                serializer_class => 'JSYNC',
            }
        );

METHODS
  get_freezer
        my $freezer = ::JSYNC->get_freezer( $column, $info, $args );
        my $string = $freezer->( $object );
        # $data isa string

  get_unfreezer
        my $unfreezer = ::JSYNC->get_unfreezer( $column, $info, $args );
        my $object = $unfreezer->( $string );

AUTHOR
    Kent Fredric <kentnl@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2014 by Kent Fredric
    <kentfredric@gmail.com>.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Release history for DBIx-Class-InflateColumn-Serializer-JSYNC

{{$NEXT}}
 - Bugfix: Avoid test failures with -Ddefault_inc_excludes_dot

 [Dependencies::Stats]
 - Dependencies changed since 0.002000, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +11 ↑3 -3 (recommends: +1, suggests: ↑2)
 - test: +1 ↓1 (recommends: +1 ↑1 ↓1)

0.002000 2014-10-06T02:56:09Z eb2acc9
 [00 Major - Toolkit changed]
 - This may affect hand operated downstreams.
 - ExtUtils::MakeMaker is now used instead of Module::Build
 - Toolkit/test coredeps are now only "recommended" to be upgraded to the latest version, not required.
 - https://questhub.io/realm/perl/quest/5388cc12bbd0be5b2500011f

 [Dependencies::Stats]
 - Dependencies changed since 0.001000, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +48 -2 (recommends: -1, suggests: +1 ↑1)
 - runtime: -1
 - test: +10 ↓1 -2 (recommends: +3)

 [Metadata]
 - move to kentnl org
 - set author id to cpan.org

 [Tests]
 - Very basic behaviour test added.

0.001000 2013-02-17T20:02:31Z
 - First version.


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "File::Find" => 0,
    "File::Temp" => 0,
    "Module::Build" => "0.4003",
    "Test::More" => "0.98"
  },
  "configure_requires" => {
    "Module::Build" => "0.4003"
  },
  "dist_abstract" => "Basic JSON Object Serialization Support for DBIx::Class.",
  "dist_author" => [
    "Kent Fredric <kentfredric\@gmail.com>"
  ],
  "dist_name" => "DBIx-Class-InflateColumn-Serializer-JSYNC",
  "dist_version" => "0.001000",
  "license" => "perl",
  "module_name" => "DBIx::Class::InflateColumn::Serializer::JSYNC",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Carp" => 0,
    "DBIx::Class::InflateColumn::Serializer" => 0,
    "JSYNC" => 0,
    "Pegex::Base" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => []
);


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


lib/DBIx/Class/InflateColumn/Serializer/JSYNC.pm
use 5.006;    # our
use strict;
use warnings;

package DBIx::Class::InflateColumn::Serializer::JSYNC;

our $VERSION = '0.002000';

# ABSTRACT: Basic JSON Object Serialization Support for DBIx::Class.

our $AUTHORITY = 'cpan:KENTNL'; # AUTHORITY

use JSYNC;
use Carp qw( croak );

sub get_freezer {
  my ( undef, undef, $col_info, undef ) = @_;
  if ( defined $col_info->{'size'} ) {
    my $size = $col_info->{'size'};
    return sub {
      my $v = JSYNC::Dumper->new()->dump( $_[0] );
      croak('Value Serialization is too big')
        if length($v) > $size;
      return $v;
    };
  }
  return sub {
    return JSYNC::Dumper->new()->dump( $_[0] );
  };
}

sub get_unfreezer {
  return sub {
    return JSYNC::Loader->new()->load( $_[0] );
  };
}

1;

__END__

=pod

=encoding UTF-8

=head1 NAME

DBIx::Class::InflateColumn::Serializer::JSYNC - Basic JSON Object Serialization Support for DBIx::Class.

=head1 VERSION

version 0.002000

=head1 DESCRIPTION

This is basically the only serialization backend I could find that wasn't "Dumper()",
and actually seemed to work with arbitrary C<bless()>

    package Foo::Result::Thing;
    __PACKAGE__->load_components('InflateColumn::Serializer', 'Core');
    __PACKAGE__->table('thing');

    ....

    __PACKAGE__->add_column(
        data => {
            data_type => 'text',
            serializer_class => 'JSYNC',
        }
    );

=head1 METHODS

=head2 get_freezer

    my $freezer = ::JSYNC->get_freezer( $column, $info, $args );
    my $string = $freezer->( $object );
    # $data isa string

=head2 get_unfreezer

    my $unfreezer = ::JSYNC->get_unfreezer( $column, $info, $args );
    my $object = $unfreezer->( $string );

=head1 AUTHOR

Kent Fredric <kentnl@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2014 by Kent Fredric <kentfredric@gmail.com>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.020.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker ;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Basic JSON Object Serialization Support for DBIx::Class.",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "DBIx-Class-InflateColumn-Serializer-JSYNC",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "DBIx::Class::InflateColumn::Serializer::JSYNC",
  "PREREQ_PM" => {
    "Carp" => 0,
    "DBIx::Class::InflateColumn::Serializer" => 0,
    "JSYNC" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "DBD::SQLite" => 0,
    "DBI" => 0,
    "DBIx::Class::Core" => 0,
    "DBIx::Class::InflateColumn::Serializer" => 0,
    "DBIx::Class::Schema" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Path::Tiny" => 0,
    "Test::Fatal" => 0,
    "Test::More" => "0.89",
    "parent" => 0
  },
  "VERSION" => "0.002000",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "DBD::SQLite" => 0,
  "DBI" => 0,
  "DBIx::Class::Core" => 0,
  "DBIx::Class::InflateColumn::Serializer" => 0,
  "DBIx::Class::Schema" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "JSYNC" => 0,
  "Path::Tiny" => 0,
  "Test::Fatal" => 0,
  "Test::More" => "0.89",
  "parent" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



